server/controllers/chartsController.js

```diff
@@ -145,14 +145,16 @@ export const graphDataController = async (dataDb, userAccess, chart_id) => {
     chart.fieldLabelValueMap.forEach((fieldLabelValueMap) => {
       const { color, label, value, targetValues } = fieldLabelValueMap
       const targetValue = targetValues[study]
-      const hasValue = subjectDayData.some(
-        (dayData) => dayData[chart.variable] == value
-      )
+      const isVariableValueEmpty = value === ''
+      const shouldCountSubject = isVariableValueEmpty
+        ? subjectDayData.every((day) => day[variable] === value)
```

Comment:
I think we're back at the original issue. We want to also count subjects when `subjectDayData` is empty.

Reply:
oh actually I missed your previous comment so it seems this will handle the case of the array being empty ✅